projects/stef.md

```diff
@@ -0,0 +1,471 @@
+  Arrow.
+- Explore the limitations of Otel/STEF, the impact of its statefulness and 
+  persistent connection on load balancers and servers that accept a large number of 
+  connections.
+- Refine and stabilize Otel/STEF specification and encourage its implementation
+  inside or outside Otel.
+- Evaluate additional use-cases that can be served by Otel/STEF, for example
+  direct-export from SDKs to a local file.
+
+Project non-goals:
+
+- We do not plan to offer Otel/STEF as a general-purpose replacement for OTLP or for Otel
+  Arrow.
+
+Unlike
+[OTLP](https://github.com/open-telemetry/opentelemetry-proto/blob/main/docs/design-goals.md),
+for Otel/STEF it is not a goal to be a universal protocol useful for all
+node types. We do not plan it to be used by instrumented applications 
+and thus don't need Otel/STEF to be implemented in all language SDKs. This 
+significantly reduces the total engineering cost of this project since we
+only need the Go implementation for the Otel Collector.
+
+Otel/STEF design takes into account learnings from Otel Arrow and borrows some of the
+technics (e.g. dictionary encoding and gRPC streams).
+
+## Benchmarks
+
+A fairly full-featured, non-production grade prototype Otel/STEF implementation for 
+metric data exists in Go. It demonstrates wire-size and performance improvements over 
+OTLP and Otel Arrow.
+
+A small subset of existing benchmarks is reproduced in this section. For more detailed 
+speed and size tests and comparison to other protocols see
+[benchmarks](https://github.com/tigrannajaryan/stef/tree/main/benchmarks).
+
+The benchmarks are run on the highest compression / slowest speed option.
+
+### Wire Size
+
+The following benchmarks show zstd-compressed wire sizes of OTLP vs Otel/STEF vs Otel 
+Arrow Streaming Mode. Benchmarks are run using
+[Otel Arrow bench](https://github.com/open-telemetry/otel-arrow), with Otel/STEF
+prototype implementation added for comparison.
+
+Dataset `hipstershop_metrics.pb`:
+
+| BATCH SIZE         | OTLP                    | Otel/STEF                       | OTEL ARROW + Stream Mode        |
+|--------------------|-------------------------|---------------------------------|---------------------------------|
+| Compressed (bytes) |                         |                                 |                                 |
+| batch_size: 128    | 4579  (total: 1.6 MB)   | 947 (x  4.83) (total: 323 kB)   | 2067 (x  2.22) (total: 705 kB)  |
+| batch_size: 1024   | 24764  (total: 1.1 MB)  | 5622 (x  4.40) (total: 242 kB)  | 10773 (x  2.30) (total: 463 kB) |
+| batch_size: 2048   | 39325  (total: 865 kB)  | 9209 (x  4.27) (total: 203 kB)  | 17808 (x  2.21) (total: 392 kB) |
+| batch_size: 4096   | 64824  (total: 713 kB)  | 15501 (x  4.18) (total: 170 kB) | 29421 (x  2.20) (total: 324 kB) |
+| batch_size: 16384  | 196877  (total: 591 kB) | 38376 (x  5.13) (total: 115 kB) | 86299 (x  2.28) (total: 259 kB) |
```

Comment:
What do you think about updating these to show the result of using OTLP + various compression algorithms. gzip is the default, but I [wrote benchmarks](https://github.com/open-telemetry/opentelemetry-collector/tree/main/config/configgrpc) for a variety of others and had good results. I wonder how Otel/STEF And OTEL Arrow + Stream Mode stack up.